qgepqwat2ili/tests/test_qgep.py

```diff
@@ -117,6 +117,36 @@ def test_case_c_import_complete_xtf_to_qgep(self):
         self.assertEqual(session.query(QGEP.manhole).get("ch080qwzNS000113").year_of_construction, 1950)
         session.close()
 
+    def test_case_d_import_complete_xtf_to_qgep(self):
```

Comment:
Rather than doing a new test case, it would be better to fix the `test_case_c_import_complete_xtf_to_qgep`.

If we still want to test behavior for invalid files (which would be nice), we could do a specific test case for that (once that specifically asserts we get a validation error, instead of just checking that the test fails)

Reply:
> I added the logs to see what kind of errors are really in the files.

Ok but since they are generated files, it's better not to version them in git.

> We have tools like ilivalidator to test xtf files - why duplicate this here?

The goal is not to duplicate ilivalidator, just to assert qgepqwat2ili handles invalid files gracefully (ie it doesn't just crash with some random messages, but instead clearly says that input is invalid).

> For me the goal of these tests are still not clear. What do they prevent? They are just applied on the testdata? Or are they used on other files e.g. the files we create or import?

These are just automated tests. It's only used in github actions to test that pull requests don't break working workflows. This code is not used at all when users actually use the library.